gateway/gateway.go

```diff
@@ -246,56 +283,62 @@ func (g *Gateway) SessionID() string {
 	return g.sessionID
 }
 
-// Reconnect tries to reconnect until the ReconnectTimeout is reached, or if
+// reconnect tries to reconnect until the ReconnectTimeout is reached, or if
```

Comment:
This is breaking the API; why the change?

Reply:
There are also changes on the contextful version of this method.

Reply:
Because `Reconnect` does not work anymore, since Closures, as discussed, invalidate the session ID.

Reply:
Closes aren't specifically relevant in the context of the caller; the function should automatically handle closing (ungracefully) and resume as normally.

---

For the record, I still think there should be a way to close without invalidating the session ID. I think a new method, maybe `Terminate`, should be added. The old `CloseGracefully` can call `Close` but marked as deprecated.

I'm not for breaking the API without bumping the major version.

Reply:
Alright, will implemt this tomorrow.

Reply:
However, I think `Pause` would be more self-explanatory, as it emphasizes that the connection can be resumed, which not be so apparent when using `Terminate`. What do you think?

Reply:
Sounds good to me.